dojo/data_store.py

```diff
@@ -0,0 +1,119 @@
+import threading
+import logging
+import os
+import csv
+import json
+import pickle
+
+logger = logging.getLogger(__name__)
+
+class DataStore:
+    _instance = None
+    _lock = threading.Lock()
+    _is_loaded = False
+    
+    def __new__(cls):
+        if cls._instance is None:
+            with cls._lock:
+                if cls._instance is None:
+                    cls._instance = super().__new__(cls)
+                    cls._instance.data = {}
+        return cls._instance
+        
+        
+    def load_file(self, file_path):
```

Comment:
We should move this processing to the `init` container, then dump the data in a Pickle (or some other efficient format) for faster loading.

We need to evaluate the memory consumption of storing this in memory, as the memory cost will incur for each UWSGI process.  If the memory cost turns out to be significant, e.g., more than 200MiB, we should instead store the information on Redis (so we pay the storage cost only once).

Reply:
https://github.com/iago-r/django-DefectDojo/pull/8